ext/fasttext/extconf.rb

```diff
@@ -1,7 +1,12 @@
 require "mkmf-rice"
 
 # -pthread and -O3 set by default
-$CXXFLAGS << " -std=c++17 $(optflags) -funroll-loops " << with_config("optflags", "-march=native")
```

Comment:
Let's only change the default on this line so `optflags` can still be passed. Also, for the arm64 check, let's use `RbConfig::CONFIG["host_cpu"] =~ /arm/i` so it's consistent with my other ML gems.

Reply:
By only the default, do you mean still pass ``optflags``, and only conditionally pass ``-march=native``?

Reply:
Yeah, I think we can do something like:

```ruby
default_optflags = RbConfig::CONFIG["host_os"] =~ /darwin/i && RbConfig::CONFIG["host_cpu"] =~ /arm/i ? "-march=armv8-a" : "-march=native"
... << with_config("optflags", default_optflags)
```